api/hypershift/v1beta1/azure.go

```diff
@@ -479,6 +479,49 @@ type AzureResourceManagedIdentities struct {
 	// +required
 	DataPlane DataPlaneManagedIdentities `json:"dataPlane"`
 }
+type AzureWorkloadIdentities struct {
+	// imageRegistryClientID is the client ID of a federated managed identity, associated with cluster-image-registry-operator, used in
+	// workload identity authentication.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12."
+	// +kubebuilder:validation:MaxLength=255
```

Comment:
we might want to have a dedicated workloadIdentity struct with the a field for clientID.
Let's not block the PR on this for the sake of moving forward for the managed changes. We will keep iterating for self managed